File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sum =0;
    void solve(TreeNode* r,int st)
    {
        if(r==NULL)
            return;
        int d=st*10+r->val;
        if(r->left==NULL&&r->right==NULL)
            sum+=d;
        solve(r->left,d);
        solve(r->right,d);
    }
    int sumNumbers(TreeNode* root) {
        solve(root,0);
        return sum;
    }
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    int sum=0;
    public void solve(TreeNode r,int l)
    {
        if(r==null)
            return;
        int s=l*10+r.val;
        if(r.left==null&&r.right==null)
            sum+=s;
        solve(r.left,s);
        solve(r.right,s);
    }
    public int SumNumbers(TreeNode root) {
        solve(root,0);
        return sum;
    }
}